main.go

```diff
@@ -295,6 +301,40 @@ func reconcileBalancer(ctx context.Context, bal balancer.Balancer, c cluster.Clu
 	}
 }
 
+func handleClusterEvents(ctx context.Context, mapic mistapiconnector.IMac, c cluster.Cluster) error {
+	eventCh := c.EventChan()
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case e := <-eventCh:
+			processClusterEvent(mapic, e)
+		}
+	}
+}
+
+func processClusterEvent(mapic mistapiconnector.IMac, e serf.UserEvent) {
+	type EventPayload struct {
+		Resource   string `json:"resource"`
+		PlaybackID string `json:"playback_id"`
+	}
+
+	go func() {
+		var eventPayload EventPayload
+		err := json.Unmarshal(e.Payload, &eventPayload)
+		if err != nil {
+			glog.Errorf("cannot unmarshall received serf event: %v", e)
```

Comment:
Should it `return` here?

Reply:
Good point. Added in 6a732de290b2faa93d747fe43e75914c362fbafd